packages/wallet/ui/src/components/Loading.jsx

```diff
@@ -0,0 +1,47 @@
+import Typography from '@mui/material/Typography';
+import CircularProgress from '@mui/material/CircularProgress';
+import Paper from '@mui/material/Paper';
+
+import { withApplicationContext } from '../contexts/Application';
+
+const Loading = ({ wantConnection, walletConnection, ...rest }) => {
+  let body;
+  if (walletConnection) {
+    const { url } = walletConnection || {};
```

Comment:
this is guaranteed truthy by line 9
```suggestion
    const { url } = walletConnection;
```